### ISSUE TITLE
StaticPaths middleware

### ISSUE THREAD

**Issue by crookse** [drashland/deno-drash-middleware#97]
## Summary

What:

Remove the static paths middleware from deno-drash and make it useable as middleware.

Why:

Serving static paths is not part of Drash's core.

## Acceptance Criteria

Below is a list of tasks that must be completed before this issue can be closed.

- [ ] Write migration guide
- [ ] Write documentation
- [ ] Write unit tests
- [ ] Write integration tests

**Comment by ebebbington** [comment]
As before request middleware is called without a response, this middleware will have to be after request middleware, which is a problem, because essentially none of the drash logic should need to run as when the request hits the web server, this middleware is going to return a response very early. for example:

**How it would work now**
1. Drash gets a request
2. runs all the logic around if request is for a resource etc
3. runs the after request middleware

**How it should work**
1. Drash gets request
2. calls before request middleware, passing in response object
3. this middleware runs, checks if request is for a static asset
4. if it is, sends the response the client
5. drash doesnt need to run any other logic, the middleware handles it all in this scenario

**Comment by ebebbington** [comment]
I understand that the response param in the middleware being a way to tell us if the request is before or ater, but maybe we should add an extra param instead so the response is always passed in

```
type MiddlewareFunction = (request: ..., response: ..., type: "before" | "after") => void
```

**Comment by crookse** [comment]
we could add an extra param so the response is always passed in, but i can't remember why we didn't do that

**Comment by crookse** [comment]
> As before request middleware is called without a response, this middleware will have to be after request middleware, which is a problem, because essentially none of the drash logic should need to run as when the request hits the web server, this middleware is going to return a response very early. for example:
> 
> **How it would work now**
> 
> 1. Drash gets a request
> 2. runs all the logic around if request is for a resource etc
> 3. runs the after request middleware
> 
> **How it should work**
> 
> 1. Drash gets request
> 2. calls before request middleware, passing in response object
> 3. this middleware runs, checks if request is for a static asset
> 4. if it is, sends the response the client
> 5. drash doesnt need to run any other logic, the middleware handles it all in this scenario

you're right. damnit. so drash short-circuits all of the logic based on the request extension. if it has a .txt or .png or .jpg or whatever, then it knows the request is for a static asset. from there it goes "alrigh we don't have to execute the resource logic... just short-circuit this request and send the static asset". however, we don't have that kind of logic in middleware, so we'd need to add that logic... and it would possibly be a breaking change.

**Comment by ebebbington** [comment]
related PR is: https://github.com/drashland/deno-drash/pull/503

we'd have to use response.send() in this middleware to end the lifecycle

But we're in talks of removing the need for servestatic and using a resource inctead, so this comment might make no difference, but its here for reference regardless
